core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -198,15 +198,15 @@ class BaseUnit : INamed, IConstruction {
         var XP = cityConstructions.getBuiltBuildings().sumBy { it.xpForNewUnits }
 
 
-        for (unique in cityConstructions.cityInfo.cityConstructions.builtBuildingUniqueMap
+        for (unique in cityConstructions.builtBuildingUniqueMap
```

Comment:
Nice catch :) Wonder what the history behind this was

Reply:
2020-08-05: `for (unique in construction.cityInfo.civInfo.getMatchingUniques2("New military units start with [] Experience"))`
2020-08-06: `for (unique in construction.cityInfo.civInfo.getMatchingUniques("New military units start with [] Experience"))`
2020-08-14: `for (unique in civInfo.getMatchingUniques("New military units start with [] Experience"))`
2020-08-18: `for (unique in construction.cityInfo.cityConstructions.builtBuildingUniqueMap.getUniques("All newly-trained [] units in this city receive the [] promotion")) {`
... @HadeanLake dunnit

Reply:
Corrected history excerpt:
2020-08-14: `if (unit.type in listOf(UnitType.Melee,UnitType.Mounted,UnitType.Armor) && construction.cityInfo.containsBuildingUnique("All newly-trained melee, mounted, and armored units in this city receive the Drill I promotion")`
2020-08-18: +`for (unique in construction.cityInfo.cityConstructions.builtBuildingUniqueMap.getUniques("All newly-trained [] units in this city receive the [] promotion")) {`
    ... replacing `cityInfo.containsBuildingUnique` with `cityInfo.cityConstructions.builtBuildingUniqueMap.getUniques` without realizing that `cityInfo` in that case came from `cityConstructions` ...
2020-10-14: +`for (unique in construction.cityInfo.cityConstructions.builtBuildingUniqueMap.getUniques("New [] units start with [] Experience in this city")`
    ... simply copied ...